### ISSUE TITLE
Add bboxes as a prompt

### ISSUE THREAD

**Issue by earzamastsev** [Gy920/segment-anything-2-real-time#2]
Supercool solution for real time streaming and object tracking! But if you add bboxes as a prompt - it will be awesome!  And methon add_new_points() is deprecated. Need to use add_new_points_or_box().

**Comment by Gy920** [comment]
> Supercool solution for real time streaming and object tracking! But if you add bboxes as a prompt - it will be awesome! And methon add_new_points() is deprecated. Need to use add_new_points_or_box().

Thank you for your suggestion. I have added an interface to use bbox as prompts. The specific usage is described [here](https://github.com/Gy920/segment-anything-2-real-time/blob/2b41ace144463ce12b0d93dad07bf255524f7e1f/demo/demo.py#L57).

**Comment by earzamastsev** [comment]
Thanks a lot! One more question: if any new object will appear in frame, how to add him to tracking with previous tracked objects?

**Comment by Gy920** [comment]
In fact, with the current pipeline, you'll need to reset the tracker and re-add prompts for both the old and new objects to track both of them. The prompt for the old object can be obtained from the tracking results of the previous frame.

**Comment by aenoca** [comment]
I find the current bounding box prompting is wrong. I instead propose to replace the lines 205 to 219 of `sam2/sam2_camera_predictor.py` with 

``
        if points is None:
            points = torch.zeros(0, 2, dtype=torch.float32)
        elif not isinstance(points, torch.Tensor):
            points = torch.tensor(points, dtype=torch.float32)
        if labels is None:
            labels = torch.zeros(0, dtype=torch.int32)
        elif not isinstance(labels, torch.Tensor):
            labels = torch.tensor(labels, dtype=torch.int32)
        if points.dim() == 2:
            points = points.unsqueeze(0)  # add batch dimension
        if labels.dim() == 1:
            labels = labels.unsqueeze(0)  # add batch dimension

        if bbox is not None:
            if not isinstance(bbox, torch.Tensor):
                    bbox = torch.tensor(bbox, dtype=torch.float32, device=points.device)
                    box_coords = bbox.reshape(1, 2, 2)
                    box_labels = torch.tensor([2, 3], dtype=torch.int32, device=labels.device)
                    box_labels = box_labels.reshape(1, 2)
                    points = torch.cat([box_coords, points], dim=1)
                    labels = torch.cat([box_labels, labels], dim=1)
``

**Comment by Gy920** [comment]
> I find the current bounding box prompting is wrong. I instead propose to replace the lines 205 to 219 of `sam2/sam2_camera_predictor.py` with
> 
> ```
>   if points is None:
>       points = torch.zeros(0, 2, dtype=torch.float32)
>   elif not isinstance(points, torch.Tensor):
>       points = torch.tensor(points, dtype=torch.float32)
>   if labels is None:
>       labels = torch.zeros(0, dtype=torch.int32)
>   elif not isinstance(labels, torch.Tensor):
>       labels = torch.tensor(labels, dtype=torch.int32)
>   if points.dim() == 2:
>       points = points.unsqueeze(0)  # add batch dimension
>   if labels.dim() == 1:
>       labels = labels.unsqueeze(0)  # add batch dimension
>   if bbox is not None:
>       if not isinstance(bbox, torch.Tensor):
>               bbox = torch.tensor(bbox, dtype=torch.float32, device=points.device)
>               box_coords = bbox.reshape(1, 2, 2)
>               box_labels = torch.tensor([2, 3], dtype=torch.int32, device=labels.device)
>               box_labels = box_labels.reshape(1, 2)
>               points = torch.cat([box_coords, points], dim=1)
>               labels = torch.cat([box_labels, labels], dim=1)
> ```

Thank you for pointing that out. I have corrected the error.
